ui/src/pages/version/components/forms/components/ResourcesPanel.js

```diff
@@ -242,16 +244,18 @@ export const ResourcesPanel = ({
           <EuiDualRange
             fullWidth
             min={0}
-            max={maxAllowedReplica}
+            max={maxAllowedReplica || 0}
             showInput
             showLabels
             value={[
-              resourcesConfig.min_replica || 0,
-              resourcesConfig.max_replica || 0,
+              !!environment ? (resourcesConfig.min_replica || 0) : 0,
```

Comment:
Why check for `!!environment` here?
Also, I think you can simplify it a little bit: `!!environment ? resourcesConfig.min_replica : 0`

Reply:
It's basically to check if an `environment` is selected or not. If it isn't, we'll display the min/max replicas a 0, otherwise we'll display the min/max replica values that's set in `resourcesConfig`.

I don't think it works the same way though 🤔:
```js
> true ? (null || 0) : 0
< 0

> true ? null : 0
< null
```

Reply:
this is to prevent `resourcesConfig.min_replica` to be used when environment is empty and when resourcesConfig.min_replica is undefined?

Reply:
Yep that's right, though I'm not 100% certain what the situations are when `resourcesConfig.min_replica` would be undefined.

Reply:
Hmm I doubt so, `resourcesConfig` is either defined as the default config that's created [here](https://github.com/caraml-dev/merlin/blob/0a79211db6ed60b266ee75cf03088b4f22eb037d/ui/src/pages/version/deploy/DeployModelVersionView.js#L76) or set to the previous deployed version endpoint, in which case we can simplify the statement to what Arief had suggested.